DynamicModeler/Logic/vtkSlicerDynamicModelerRevolveTool.cxx

```diff
@@ -0,0 +1,375 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, Cancer
+  Care Ontario, OpenAnatomy, and Brigham and Women's Hospital through NIH grant R01MH112748.
+
+==============================================================================*/
+
+#include "vtkSlicerDynamicModelerRevolveTool.h"
+
+#include "vtkMRMLDynamicModelerNode.h"
+
+// MRML includes
+#include <vtkMRMLMarkupsFiducialNode.h>
+#include <vtkMRMLMarkupsLineNode.h>
+#include <vtkMRMLMarkupsPlaneNode.h>
+#include <vtkMRMLMarkupsAngleNode.h>
+#include <vtkMRMLModelNode.h>
+#include <vtkMRMLTransformNode.h>
+
+// VTK includes
+#include <vtkCommand.h>
+#include <vtkGeneralTransform.h>
+#include <vtkIntArray.h>
+#include <vtkFloatArray.h>
+#include <vtkPointData.h>
+#include <vtkRotationalExtrusionFilter.h>
+#include <vtkPolyDataNormals.h>
+#include <vtkSmartPointer.h>
+#include <vtkStringArray.h>
+#include <vtkTransform.h>
+#include <vtkTransformPolyDataFilter.h>
+#include <vtkTriangleFilter.h>
+#include <vtkFeatureEdges.h>
+#include <vtkAppendPolyData.h>
+
+//----------------------------------------------------------------------------
+vtkToolNewMacro(vtkSlicerDynamicModelerRevolveTool);
+
+const char* REVOLVE_INPUT_MODEL_REFERENCE_ROLE = "Revolve.InputModel";
+const char* REVOLVE_INPUT_MARKUPS_REFERENCE_ROLE = "Revolve.InputMarkups";
+const char* REVOLVE_OUTPUT_MODEL_REFERENCE_ROLE = "Revolve.OutputModel";
+const char* REVOLVE_ANGLE_DEGREES = "Revolve.AngleDegrees";
+const char* REVOLVE_AXIS_AT_ORIGIN = "Revolve.AxisAtOrigin";
```

Comment:
Origin of what? The name should reflect that.

Reply:
It is the rotation axis' origin, maybe could call it `OriginOfRotationAxis`

Reply:
Isn't the rotation axis already going through its origin? Do you mean that the rotation axis position is shifted to the world coordinate system's origin?

Reply:
Consider it works like this:
```
transform.PostMultiply()
transform.Translate(-origin) # origin is first point of a markupsLine or a markupsPlane
transform.RotateWXYZ(angleInDegrees, axisOfRotation)
transform.Translate(origin)
```
Does it clarify?

Reply:
Changed it to `const char* REVOLVE_AXIS_IS_AT_ORIGIN = "Revolve.AxisIsAtOrigin";`

Reply:
We are getting closer, but I still don't know what origin this refers to. Origin of the axis? But of course we use the origin of the axis, what else would we use as location of the rotation axis if not rotation axis?

Would we use the world coordinate system origin? But then why don't we place the markup in the origin of the world coordinate system?